RadClass/RadClass.py

```diff
@@ -122,8 +124,14 @@ def collect_rows(self):
                     len(self.processor.timestamps) - 1)
 
         # enumerate number of rows to integrate exclusive of the endpoint
-        rows = np.arange(start_i, end_i)
-        return rows
+        rows_idx = np.arange(start_i, end_i)
+
+        # check if rows are stored in cache
+        if np.count_nonzero(np.isin(self.cache_idx,
```

Comment:
Can't we just compare the end of `cache_idx` to `end_i` somehow?  This seems bulky and complicated? Maybe I'm missing something?

Reply:
This ensures that *every* row (or row index) is stored in cache, but maybe that's too exhaustive and just checking for the end is sufficient. How about this?

```
if end_i not in self.cache_idx:
     self.run_cache()
```

Reply:
I agree that your original case is more robust, but I think you do everything linearly, so this proposal should be enough